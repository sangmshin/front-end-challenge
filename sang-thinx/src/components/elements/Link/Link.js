import React from 'react';
import { func, node, string } from 'prop-types';

export const Link = ({
  accessibleText,
  children,
  className,
  onClick,
  target,
  to,
}) => {
  if (!children) return null;

  return (
    <a
      href={to}
      className={className}
      onClick={onClick}
      target={target}
    >
      {children}
      {accessibleText && <span className='accessible'>{accessibleText}</span>}
    </a>
  )
}

Link.propTypes = {
  accessibleText: string,
  children: node.isRequired,
  className: string,
  onClick: func,
  target: string,
  to: string,
}

Link.defaultProps = {
  accessibleText: '',
  className: null,
  onClick: _ => _,
  target: null,
  to: null,
}

export default Link;