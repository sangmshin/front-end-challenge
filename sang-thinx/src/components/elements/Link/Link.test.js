import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Link } from './Link';

describe('Component :: Link', () => {
  describe('#render', () => {
    const component = shallow(
      <Link>
        <p>shethinx.com</p>
      </Link>
    )
    it('should match snapshot', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
