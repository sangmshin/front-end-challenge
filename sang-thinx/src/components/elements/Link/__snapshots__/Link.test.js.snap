// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: Link #render should match snapshot 1`] = `
<a
  className=""
  href={null}
  onClick={[Function]}
  target={null}
>
  <p>
    shethinx.com
  </p>
</a>
`;
